internal/app/machined/internal/server/v1alpha1/v1alpha1_monitoring.go

```diff
@@ -19,7 +19,7 @@ import (
 
 // Hostname implements the machine.MachineServer interface.
 func (s *Server) Hostname(ctx context.Context, in *empty.Empty) (*machine.HostnameResponse, error) {
-	hostname, err := os.Hostname()
+	hostname, err := s.Controller.Runtime().NodeName()
```

Comment:
I'm not _sure_ that we need the nodename here. @smira maybe you know?

Reply:
This is only used in monitoring APIs, e.g. dashboard. I would rather leave it as os.Hostname, as it's Talos hostname, not Kubernetes node name

Reply:
cool, yeah that's what I thought. updated in amended commit to roll that back.